managed-pointer/cpp-20/D0567R1.md

```diff
@@ -5,222 +5,384 @@
 
-*Figure 2: Extensions to unified interface for execution*
+#### 3.3.3. Memory accessibiity
 
-As described previously there are three ways in which a synchronisation operation can be triggered, and these are separated into explicit and implicit.
+During the lieftime of a `managed_ptr` it's managed memory allocation can exist in the memory region(s) associated with any number of *execution contexts*, but may only be accessible in one of these memory regions at any given time. The *execution context* who's memory region is currently accessible is said to be the accessible *execution context*.
 
-The first is by calling one of the member functions or global customisation point functions described above. This will trigger a synchronisation operation to the memory region of the **execution context** associated with the **executor**. This method is useful as it allows finer control over the synchronisation operations by chaining continuations. This is implicit.
+> This limited may be relaxed in a future version of the paper when a `managed_ptr` can be concurrently accessed in more than one memory region. In this case there could be more than one accessible memory region. This is further described in section 5.1.
 
-The second is by calling the ``get()`` member function on the ``managed_ptr`` itself as this will implicitly trigger a call to ``get()`` on the current accessible **execution context** if the host CPU is not currently accessible. This method is useful when you want to synchronise data back to the host CPU in order to use the resulting data in regular sequential code. This is explicit.
+The `managed_ptr` must maintain a reference to the current accessible *execution context*.
 
-The third is by passing a ``managed_ptr`` to an executor control structure such as ``async()``, as this will implicitly trigger a call to the ``put()`` member function on the **executor**. If the host CPU is not accessible at this point then this will implicitly trigger a call to``get()`` on the ``managed_ptr`` which will subsequently implicitly trigger a call to ``get()`` on the currently accessible **execution context**. This method is useful as it allows you to simply pass a  ``managed_ptr`` directly to a control structure without requiring an explicit call to ``put()``. This is implicit.
+The associated *execution context* and *executor* of a `managed_ptr` can also be accessible, the `managed_ptr` has a member function for querying wether the associated *execution context* and *executor* is accessible; `is_accessible()`.
 
-## Examples
+The data at the memory region of the `managed_ptr`s associated *execution context* and *executor* can be read and modified (if the type `T` is a non-const type) via the de-reference and pointer operators.
```

Comment:
does that mean it is possible to get the address of that pointer? If so, is pointer arithmetic valid?